enzyme/Enzyme/MLIR/Passes/InlineEnzymeRegions.cpp

```diff
@@ -180,30 +182,42 @@ outlineAutoDiffFunc(enzyme::AutoDiffRegionOp op, StringRef funcName,
 }
 
 LogicalResult outlineEnzymeAutoDiffRegion(enzyme::AutoDiffRegionOp op,
-                                          StringRef defaultFuncName,
+                                          StringRef funcName,
                                           OpBuilder &builder) {
-  StringRef funcName = op.getFn().value_or(defaultFuncName);
   OpBuilder::InsertionGuard insertionGuard(builder);
   builder.setInsertionPointAfter(op->getParentOfType<SymbolOpInterface>());
 
   SmallVector<enzyme::Activity> argActivities =
       llvm::map_to_vector(op.getActivity().getAsRange<enzyme::ActivityAttr>(),
                           [](auto attr) { return attr.getValue(); });
-
-  size_t numPrimalArgs = op.getActivity().size();
-  SmallVector<Value> primalInputs(op.getInputs().begin(),
-                                  op.getInputs().begin() + numPrimalArgs);
-  SmallVector<Value> seedInputs(op.getInputs().begin() + numPrimalArgs,
+  size_t numSeeds = llvm::count_if(
```

Comment:
I think this can be avoided once #2495 is merged. We could just call `op.getDifferentialReturns()` to get seedInputs